glyphs-reader/src/plist.rs

```diff
@@ -472,6 +506,387 @@ impl From<BTreeMap<String, Plist>> for Plist {
+                    return Ok(());
+                }
+                loop {
+                    self.skip_rec()?;
+                    if self.consume(b')') {
+                        return Ok(());
+                    }
+                    if !self.consume(b',') {
+                        return Err(Error::ExpectedComma);
+                    }
+                }
+            }
+            _ => Err(Error::SomethingWentWrong),
+        }
+    }
+}
+
+#[derive(Debug, Default, PartialEq, TryFromPlist)]
+//#[derive(Debug, Default, PartialEq)]
+struct Features {
+    features: Vec<Feature>,
+}
+
+#[derive(Debug, Default, PartialEq, TryFromPlist)]
+struct Feature {
+    automatic: i64,
+    disabled: Option<i64>,
+    code: String,
+    name: Option<String>,
+}
+
+// Parse a vec of anything we know how to parse
+// TODO: haven't quite gotten this to work
+// impl<T> TryFrom<&crate::plist::Tokenizer<'_>> for Vec<T>
+// where
+//     for<'a> T: TryFrom<&'a Tokenizer<'a>, Error = crate::plist::Error>
+// {
+//     type Error = crate::plist::Error;
+//     fn try_from(tokenizer: &crate::plist::Tokenizer<'a>) -> Result<Self, Self::Error> {
+//         let mut list = Vec::new();
+//         if !tokenizer.consume(b'(') {
+//             return Err(Error::ExpectedOpenParen);
+//         };
+//         if tokenizer.consume(b')') {
+//             return Ok(list);
+//         }
+//         loop {
+//             list.push(tokenizer.try_into()?);
+//             if tokenizer.consume(b')') {
+//                 return Ok(list);
+//             }
+//             if !tokenizer.consume(b',') {
+//                 return Err(Error::ExpectedComma);
+//             };
+//         }
+//     }
+// }
+
+pub(crate) struct VecDelimiters {
+    start: u8,
+    end: u8,
+    sep: u8,
+}
+
+pub(crate) const CSV_IN_PARENS: VecDelimiters = VecDelimiters {
+    start: b'(',
+    end: b')',
+    sep: b',',
+};
+pub(crate) const CSV_IN_BRACES: VecDelimiters = VecDelimiters {
+    start: b'{',
+    end: b'}',
+    sep: b',',
+};
+pub(crate) const SEMICOLON_SV_IN_BRACES: VecDelimiters = VecDelimiters {
+    start: b'{',
+    end: b'}',
+    sep: b';',
+};
+
+pub(crate) fn parse_delimited_vec<T>(
+    tokenizer: &Tokenizer<'_>,
+    delim: VecDelimiters,
+) -> Result<Vec<T>, crate::plist::Error>
+where
+    for<'a> T: TryFrom<&'a Tokenizer<'a>, Error = crate::plist::Error>,
+{
+    let mut list = Vec::new();
+    if !tokenizer.consume(delim.start) {
+        return Err(Error::ExpectedChar(delim.start as char));
+    };
+    loop {
+        if tokenizer.consume(delim.end) {
+            return Ok(list);
+        }
+        list.push(tokenizer.try_into()?);
+        if tokenizer.consume(delim.end) {
+            return Ok(list);
+        }
+        if !tokenizer.consume(delim.sep) {
+            return Err(Error::ExpectedChar(delim.sep as char));
+        };
+    }
+}
+
+pub(crate) fn parse_vec<T>(tokenizer: &Tokenizer<'_>) -> Result<Vec<T>, crate::plist::Error>
```

Comment:
any reason not to have these two functions be methods on `Tokenizer`?

Reply:
That would be better!